### PR TITLE
Upgrade to GA

### PR DESCRIPTION
**Pull request by salrashid123**
* Upgrade API
* Tag vault plugin release as `1.0.3`
* Introduce `CAPool`
* Add config for
  -   "issuing_certificate_authority": (optional)
  -   "certificate_template": will only read templates in current project

### CODE REVIEW

**Comment by flukschander** [comment]
Please add the changes from PR #2 to the final/next release

File: README.md [hunk 1]
```diff
@@ -56,11 +56,11 @@ To issue certificates, you need to first define a profile (config) for the mount
 
 1. Define a config profile
 
-A profile dictates the specifications of the CA a specific Vault mount will use.  In the example used here, the mount path is `gcppca` with the CA of `prod-root`
+A profile dictates the specifications of the CA a specific Vault mount will use.  In the example used here, the mount path is `gcppca` with the CAPool of `my-pool`
```
Comment (haydentherapper) [inline README.md:59]: On line 33, can we note that you need to create both a CA pool and a CA?

File: README.md [hunk 2]
```diff
@@ -102,7 +102,7 @@ path "gcppca/issue-with-genkey/my_tls_cert_rsa_1" {
       "validity"= ["P30D"]
       "dns_san" = ["client.domain.com,client2.domain.com"]        
       "subject" = ["C=US,ST=California,L=Mountain View,O=Google LLC,CN=google.com"]
-      "reusable_config" = ["leaf-server-tls"]     
+      "issuing_certificate_authority" = ["prod-root"]
```
Comment (haydentherapper) [inline README.md:105]: Can we remove this from the default example? I think leaving the note is useful, but we'd like to encourage customers to not target specific CAs except when testing out a staged CA.

Comment (haydentherapper) [inline README.md:105]: Can we remove this from the default example? We'd like to encourage customers to not target specific CAs except when testing out a staged CA.

File: README.md (hunk 1 above)
Comment (haydentherapper) [inline README.md:59]: On line 33, can we note that you need to create both a CA pool and a CA?

File: README.md [hunk 3]
```diff
@@ -81,15 +81,15 @@ Under no circumstance does this plugin retain the private key for any certificat
...
 
 Deleting the key in Vault will revoke the certificate in CA Service which also means the same name cannot be reused.
 
 ### Vault Generated
 
 To generate a certificate keypair on vault, first apply a configuration that allows Vault to reference which CA to sign against 
 
-The configuration below will generate a certificate called `my_tls_cert_rsa_1` within CA Service using a GCP CA `prod-root` that was defined earlier by specifying `gcppca/config`.
+The configuration below will generate a certificate called `my_tls_cert_rsa_1` within CA Service using a GCP CA `prod-root` that was defined separately.  Note, `issuing_certificate_authority` is an optional parameter which if omitted, will cause the certificate to get issued by any CA in the pool.
```
Comment (haydentherapper) [inline README.md:92]: Can we remove the 'note'? As mentioned below, we'd like to encourage customers to not target specific CAs. I see that there is documentation for this parameter already.

File: README.md [hunk 4]
```diff
@@ -196,6 +196,7 @@ Plugin configuration supports various options that are common and mode-specific
 |:------------|-------------|
 | **`validity`** | `string` validity of the issued certificate (default: `P30d`) |
 | **`labels`** | `[]string` list of GCP labels to apply to the certificate (format `k1=v1,k2=v2`) |
+| **`issuing_certificate_authority`** | `string` Optional. The resource ID of the CertificateAuthority that should issue the certificate. |
```
Comment (haydentherapper) [inline README.md:199]: Can we add:
"By default, the certificate will be issued from any of the active CAs in the CA Pool."

File: README.md [hunk 5]
```diff
@@ -204,7 +205,7 @@ Plugin configuration supports various options that are common and mode-specific
 | **`key_type`** | `string` what type of key to generate (default: `rsa`; either `rsa` or `ecdsa`; cannot be specified if `csr` is set) |
 | **`key_usage`** | `[]string` what are the `key_usage` settings (default: `[]`) |
 | **`extended_key_usage`** | `[]string` what are the `extended_key_usage` settings (default: `[]`)   |
-| **`reusable_config`** | `string` reusable_config to use (cannot be set if `key_usage`,`extended_key_usage` is set; default `[]`) |
+| **`certificate_template`** | `string` reusable_config to use (cannot be set if `key_usage`,`extended_key_usage` is set; default `[]`) |
```
Comment (haydentherapper) [inline README.md:208]: Change 'reusable_config' to "Certificate template"

**Review by Hayden-IO — COMMENTED** [review]
Looks good! I left a few small comments.

**Pull request merged**